doc/Develop-new-view.md

```diff
@@ -0,0 +1,183 @@
+# Developing a new view
+
+This guide will describe the mandatory and optional steps necessary to implement a new view for the VSCode Trace Viewer. It covers the full development stack, including the Trace Compass trace server, the Trace Server Protocol (TSP) and impacted libraries used by the VSCode Trace Viewer.
+
+## Data Provider
+
+If the new view is to be populated using a data model that is not already available from the trace server, then a new data provider will need to be implemented.
+
+The data provider's role is to provide a model corresponding to the requested method and parameters, possibly by scheduling an analysis and querying its resulting output, or any other method to extract data from the trace and compute the appropriate result.
```

Comment:
quibble: add a comma after "or any other method" ?  Alternatively, this is a pretty long sentence that could be broken-down into 2.